src/main/org/firebirdsql/gds/ng/wire/version13/V13Statement.java

```diff
@@ -0,0 +1,255 @@
+     * @throws IOException
+     */
+    protected RowValue readSqlData() throws SQLException, IOException {
+        final RowDescriptor rowDescriptor = getFieldDescriptor();
+        final RowValue rowValue = rowDescriptor.createDefaultFieldValues();
+        final BlrCalculator blrCalculator = getDatabase().getBlrCalculator();
+
+        synchronized (getDatabase().getSynchronizationObject()) {
+            final XdrInputStream xdrIn = getXdrIn();
+            int nullBitsLen = rowDescriptor.getCount() / 8;
+            if (nullBitsLen % 8 != 0) {
+                nullBitsLen++;
+            }
+            byte[] nullBitsBytes = new byte[nullBitsLen];
+            xdrIn.readFully(nullBitsBytes, 0, nullBitsLen);
+            BigInteger nullBits = new BigInteger(nullBitsBytes);
+
+            for (int idx = 0; idx < rowDescriptor.getCount(); idx++) {
+                final FieldDescriptor fieldDescriptor = rowDescriptor.getFieldDescriptor(idx);
+                final FieldValue fieldValue = rowValue.getFieldValue(idx);
+                if (nullBits.testBit(idx)) {
+                    fieldValue.setFieldData(null);
+                    continue;
+                }
+                int len = blrCalculator.calculateIoLength(fieldDescriptor);
+                byte[] buffer;
+                if (len == 0) {
+                    // Length specified in response
+                    len = xdrIn.readInt();
+                    buffer = new byte[len];
+                    xdrIn.readFully(buffer, 0, len);
+                    xdrIn.skipPadding(len);
+                } else if (len < 0) {
+                    // Buffer is not padded
+                    buffer = new byte[-len];
+                    xdrIn.readFully(buffer, 0, -len);
+                } else {
+                    // len is incremented in calculateIoLength to avoid value 0 so it must be decremented
+                    len--;
+                    buffer = new byte[len];
+                    xdrIn.readFully(buffer, 0, len);
+                    xdrIn.skipPadding(len);
+                }
+                fieldValue.setFieldData(buffer);
+            }
+        }
+        return rowValue;
+    }
+
+    /**
+     * Write a set of SQL data from a list of {@link FieldValue} instances.
+     *
+     * @param rowDescriptor
+     *         The row descriptor
+     * @param fieldValues
+     *         The List containing the SQL data to be written
+     * @throws IOException
+     *         if an error occurs while writing to the underlying output stream
+     */
+    protected void writeSqlData(final RowDescriptor rowDescriptor, final RowValue fieldValues) throws IOException, SQLException {
+        synchronized (getDatabase().getSynchronizationObject()) {
+            final XdrOutputStream xdrOut = getXdrOut();
+            final BlrCalculator blrCalculator = getDatabase().getBlrCalculator();
+            // null indicator bitmap
+            BigInteger nullBits = BigInteger.valueOf(0);
```

Comment:
Looking at protocol.cpp `NullBitmap`, a `BitSet` should be used instead of `BigInteger` because of the byte order (this would also simplify flipping the bits).